src/core/BaseUtilities.ts

```diff
@@ -568,17 +577,74 @@ export class BaseUtilities {
         }
     }
 
-    readonly #ensuredGuilds = new WeakSet<Guild>();
-    public async ensureMemberCache(guild: Guild): Promise<void> {
-        if (this.#ensuredGuilds.has(guild))
-            return;
+    readonly #guildsWithResolvedMembers = new WeakMap<Guild, Promise<void>>();
+    public ensureMemberCache(guild: Guild): Promise<void> {
+        let resolve = this.#guildsWithResolvedMembers.get(guild);
+        if (resolve === undefined) {
+            this.#guildsWithResolvedMembers.set(guild, resolve = this.#ensureMemberCache(guild).catch(err => {
```

Comment:
Why do you do the `resolve =` bit in here? The parameter is a Promise<void>, right? Wouldn't the rest of the line after the = be enough in this case?

Reply:
Because I need to return the promise that gets created here from this method on line 590. If I dont assign to it then id potentially be returning undefined